salt/modules/dockercompose.py

```diff
@@ -915,8 +960,11 @@ def up(path, service_names=None):
         return project
     else:
         try:
-            result = _get_convergence_plans(project, service_names)
-            ret = project.up(service_names)
+            if HAS_PYTHON_ON_WHALES:
+                project.compose.up(services=service_names)
```

Comment:
When I try this, it runs indefinitely. I am uncertain what it was before. I think it needs to be run in detached mode.
`project.compose.up(services=service_names, detach=True)`